plugins/goodix-tp/fu-goodixtp-common.c

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (C) 2020 Richard Hughes <richard@hughsie.com>
+ *
+ * SPDX-License-Identifier: LGPL-2.1+
+ */
+
+#include "config.h"
+
+#include <linux/hidraw.h>
+#include <linux/input.h>
+
+#include "fu-goodixtp-common.h"
+
+gboolean
+get_report(FuDevice *device, guint8 *buf, GError **error)
+{
+	guint8 rcv_buf[PACKAGE_LEN + 1] = {0};
+	gboolean ret;
+
+	rcv_buf[0] = REPORT_ID;
+	g_clear_error(error);
+	ret = fu_udev_device_ioctl((FuUdevDevice *)device,
+				   HIDIOCGFEATURE(PACKAGE_LEN),
+				   rcv_buf,
+				   NULL,
+				   GOODIX_DEVICE_IOCTL_TIMEOUT,
+				   error);
+	if (ret && rcv_buf[0] == REPORT_ID) {
+		memcpy(buf, rcv_buf, PACKAGE_LEN);
+		return TRUE;
+	}
```

Comment:
please do the opposite, i.e. if fu_udev_device_ioctl fails then return FALSE, then do the success path